# Profitable App Profiles for the App Store and Google Play Markets

Our aim in this project is to find mobile app profiles that are profitable for the App Store and Google Play markets. We're working as data analysts for a company that builds Android and iOS mobile apps, and our job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build.

At our company, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means that our revenue for any given app is mostly influenced by the number of users that use our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.

Open and read dataset

In [1]:
from csv import reader


file_ios = open("AppleStore.csv", encoding="utf8")
read_cell_ios = reader(file_ios)
ios = list(read_cell_ios)
ios_header = ios[0]
ios = ios[1:]

file_android = open("googleplaystore.csv", encoding="utf8")
read_cell_android = reader(file_android)
android = list(read_cell_android)
android_header = android[0]
android = android[1:]

Create explore data function

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

Explore dataset iOS

In [3]:
explore_data(ios,0,3,True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


Explore header of dataset iOS

In [4]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


Explore dataset Android

In [5]:
explore_data(android,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


Explore header of dataset Android

In [6]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


Delete wrong row in android dataset missing 'Rating'

In [7]:
del android[10472]

explore_data(android,0,0,True)

Number of rows: 10840
Number of columns: 13


## Removing Duplicate Entries

- Create a function to search for duplicates by name in our data
- Find out the number of duplicates and unique applications in each dataset (Yes, iOS too. 2 duplicates it 2 duplicates)

In [8]:
def duplicate_data (data, index):
    duplicate_apps = []
    unique_apps = []
    for app in data:
        name = app[index]
        if name in unique_apps:
            duplicate_apps.append(name)
        else:
            unique_apps.append(name)
    return duplicate_apps


print('Number of Android apps:', len(android))
print('Number of duplicate Android apps:', len(duplicate_data(android,0)))
print('Number of unique Android apps:', len(android)-len(duplicate_data(android,0)))
print('\n')
print('Number of iOS apps:', len(ios))
print('Number of duplicate iOS apps:', len(duplicate_data(ios,1)))
print('Number of unique Android apps:', len(ios)-len(duplicate_data(ios,1)))


Number of Android apps: 10840
Number of duplicate Android apps: 1181
Number of unique Android apps: 9659


Number of iOS apps: 7197
Number of duplicate iOS apps: 2
Number of unique Android apps: 7195


We don't remove the duplicates randomly. Criterion - rating

- Create a function to search duplicates by name and max rating
- For clear data we use criterion - rating. Do not forget that the data structure in Android and iOS is different.

In [9]:
def clear_dup_data (data, index1, index2):
    reviews_max = {}
    for app in data:
        name = app[index1]
        rating = float(app[index2])
        if name in reviews_max and reviews_max[name] < rating:
            reviews_max[name] = rating
        if name not in reviews_max:
            reviews_max[name] = rating
            
    clean = []
    added = []
    for app in data:
        name = app[index1]
        rating = float(app[index2])
        if (reviews_max[name] == rating) and (name not in added):
            clean.append(app)
            added.append(name)
    return clean

android = clear_dup_data(android,0,3)
print('Number of Android apps:', len(android))

ios = clear_dup_data(ios,1,5)
print('Number of iOS apps:', len(ios))

Number of Android apps: 9659
Number of iOS apps: 7195


Now we have 2 new dataset and number of rows the same 'unique apps' as above

### Removing Non-English Apps


In [10]:
print(ios[813][1])
print(ios[6731][1])

print(android[4412][0])
print(android[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
激ムズ！和のひとふで書き！ 〜頭をつかう脳トレパズルゲーム〜
中国語 AQリスニング
لعبة تقدر تربح DZ


Create funtion for search non-eng name apps

In [11]:
def is_eng (string):
    for c in string:
        if ord(c) > 127:
            return False
    
    return True

print(is_eng('Instagram'))
print(is_eng('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_eng('Docs To Go™ Free Office Suite'))
print(is_eng('Instachat 😜'))

True
False
False
False


If the input string has more than three characters that fall outside the ASCII range (0 - 127), then the function return False (identify the string as non-English), otherwise it return True.

In [12]:
def is_eng (string):
    n = 0
    for c in string:
        if ord(c) > 127:
            n += 1
    
    if n > 3:
        return False
    else:
        return True

print(is_eng('Instagram'))
print(is_eng('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_eng('Docs To Go™ Free Office Suite'))
print(is_eng('Instachat 😜'))

True
False
True
True


Below, we use the is_english() function to filter out the non-English apps for both data sets:

In [13]:
android_eng = []
ios_eng = []

for app in android:
    name = app[0]
    if is_eng(name):
        android_eng.append(app)
        
for app in ios:
    name = app[1]
    if is_eng(name):
        ios_eng.append(app)
        
explore_data(android_eng,0,2,True)
print('\n')
explore_data(ios_eng,0,2,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 6181
Number of columns: 16


We can see that we're left with 9614 Android apps and 6181 iOS apps.

### Isolate Free App

In [15]:
android_free = []
ios_free = []

for app in android_eng:
    price = app[7]
    if price == "0":
        android_free.append(app)

for app in ios_eng:
    price = app[4]
    if price == "0.0":
        ios_free.append(app)
        

android = clear_dup_data(android,0,3)
print('Number of Free Android apps:', len(android_free))

ios = clear_dup_data(ios,1,5)
print('Number of Free iOS apps:', len(ios_free))
    

Number of Free Android apps: 8864
Number of Free iOS apps: 3220


### Most Common Apps by Genre and Category


#### Part One
Create display freq table function

In [21]:
def freq_table(data, index):
    total = 0
    table = {}
    
    for row in data:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    total_perc = {}
    for element in table:
        total_perc[element] = (table[element]/total)*100
    return total_perc

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        value = (table[key], key)
        table_display.append(value)
    
    table_sort = sorted(table_display,reverse = True)
    for entry in table_sort:
        print(entry[1],':',entry[0])

#### Part Two
We start by examining the frequency table for the prime_genre column of the App Store data set.

In [22]:
display_table(ios_free,-5)

Games : 58.13664596273293
Entertainment : 7.888198757763975
Photo & Video : 4.968944099378882
Education : 3.6645962732919255
Social Networking : 3.291925465838509
Shopping : 2.608695652173913
Utilities : 2.515527950310559
Sports : 2.142857142857143
Music : 2.049689440993789
Health & Fitness : 2.018633540372671
Productivity : 1.7391304347826086
Lifestyle : 1.5838509316770186
News : 1.3354037267080745
Travel : 1.2422360248447204
Finance : 1.1180124223602486
Weather : 0.8695652173913043
Food & Drink : 0.8074534161490683
Reference : 0.5590062111801243
Business : 0.5279503105590062
Book : 0.43478260869565216
Navigation : 0.18633540372670807
Medical : 0.18633540372670807
Catalogs : 0.12422360248447205


The general impression is that App Store is dominated by apps that are designed for fun, while apps with practical purposes are more rare

However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.

Examining the Genres and Category columns of the Google Play data set (two columns which seem to be related).

In [23]:
display_table(android_free,1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

In [24]:
display_table(android_free,9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

### Most Popular Apps by Genre on the App Store

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

In [29]:
genres_ios = freq_table(ios_free, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_free:
        genre_app = app[-5]
        if genre_app == genre:
            rat = float(app[5])
            total += rat
            len_genre += 1
    avg_rat = total/len_genre
    print(genre,':',round(avg_rat,2))
    


Social Networking : 71548.35
Photo & Video : 28441.54
Games : 22812.92
Music : 57326.53
Reference : 74942.11
Health & Fitness : 23298.02
Weather : 52279.89
Utilities : 18684.46
Travel : 28243.8
Shopping : 26919.69
News : 21248.02
Navigation : 86090.33
Lifestyle : 16485.76
Entertainment : 14029.83
Food & Drink : 33333.92
Sports : 23008.9
Book : 39758.5
Finance : 31467.94
Education : 7003.98
Productivity : 21028.41
Business : 7491.12
Catalogs : 4004.0
Medical : 612.0


Navigation, Social Network, Photo&Video, Music have good average number because the have few giants like Google Maps, Facebook, Pinterest, Skype, Pandora, Spotify, Shazam.


In [49]:
for app in ios_free:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


### Most Popular Apps by Genre on Google Play


For the Google Play market, we actually have data about the number of installs, but its open-ended (100+, 1,000+, 5,000+, etc.):

In [37]:
display_table(android_free,5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


In [44]:
table = freq_table(android_free,1)
table_sort = []

for category in table:
    total = 0
    len_category = 0
    for app in android_free:
        category_app = app[1]
        if category_app == category:
            installs = app[5]
            installs = installs.replace(',','')
            installs = installs.replace('+','')
            total += float(installs)
            len_category += 1
    avg_installs = total / len_category
    cat_inst = (category, int(avg_installs))
    table_sort.append(cat_inst)

table_sort = sorted(table_sort, reverse = True, key = lambda t: t[1])
for row in table_sort:
    print(row[0],":",row[1])

COMMUNICATION : 38456119
VIDEO_PLAYERS : 24727872
SOCIAL : 23253652
PHOTOGRAPHY : 17840110
PRODUCTIVITY : 16787331
GAME : 15588015
TRAVEL_AND_LOCAL : 13984077
ENTERTAINMENT : 11640705
TOOLS : 10801391
NEWS_AND_MAGAZINES : 9549178
BOOKS_AND_REFERENCE : 8767811
SHOPPING : 7036877
PERSONALIZATION : 5201482
WEATHER : 5074486
HEALTH_AND_FITNESS : 4188821
MAPS_AND_NAVIGATION : 4056941
FAMILY : 3695641
SPORTS : 3638640
ART_AND_DESIGN : 1986335
FOOD_AND_DRINK : 1924897
EDUCATION : 1833495
BUSINESS : 1712290
LIFESTYLE : 1437816
FINANCE : 1387692
HOUSE_AND_HOME : 1331540
DATING : 854028
COMICS : 817657
AUTO_AND_VEHICLES : 647317
LIBRARIES_AND_DEMO : 638503
PARENTING : 542603
BEAUTY : 513151
EVENTS : 253542
MEDICAL : 120550


On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [53]:
for app in android_free:
    if app[1] == 'TOOLS' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google : 1,000,000,000+
Google Translate : 500,000,000+
Calculator : 100,000,000+
Device Help : 100,000,000+
Account Manager : 100,000,000+
SHAREit - Transfer & Share : 500,000,000+
Samsung Calculator : 100,000,000+
Gboard - the Google Keyboard : 500,000,000+
Google Korean Input : 100,000,000+
Share Music & Transfer Files - Xender : 100,000,000+
Tiny Flashlight + LED : 100,000,000+
GO Keyboard - Cute Emojis, Themes and GIFs : 100,000,000+
Speedtest by Ookla : 100,000,000+
CM Locker - Security Lockscreen : 100,000,000+
Applock : 100,000,000+
Clean Master- Space Cleaner & Antivirus : 500,000,000+
Lookout Security & Antivirus : 100,000,000+
Google Now Launcher : 100,000,000+
360 Security - Free Antivirus, Booster, Cleaner : 100,000,000+
Samsung Smart Switch Mobile : 100,000,000+
Avast Mobile Security 2018 - Antivirus & App Lock : 100,000,000+
AppLock : 100,000,000+
AVG AntiVirus 2018 for Android Security : 100,000,000+
Security Master - Antivirus, VPN, AppLock, Booster : 500,000,000+
Batt

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [55]:
for app in android_free:
    if app[1] == 'TOOLS':
        print(app[0], ':', app[5])

Google : 1,000,000,000+
Google Translate : 500,000,000+
Moto Display : 10,000,000+
Motorola Alert : 50,000,000+
Motorola Assist : 50,000,000+
Moto Suggestions ™ : 1,000,000+
Moto Voice : 10,000,000+
Calculator : 100,000,000+
Device Help : 100,000,000+
Account Manager : 100,000,000+
myMetro : 10,000,000+
File Manager : 50,000,000+
My Telcel : 50,000,000+
Calculator - free calculator, multi calculator app : 10,000,000+
ASUS Sound Recorder : 10,000,000+
iWnn IME for Nexus : 5,000,000+
Samsung Max - Data Savings & Privacy Protection : 10,000,000+
Android TV Remote Service : 1,000,000+
ZenUI Help : 10,000,000+
Calculator - free calculator ,multi calculator app : 100,000+
SHAREit - Transfer & Share : 500,000,000+
ZenUI Keyboard – Emoji, Theme : 10,000,000+
Files Go by Google: Free up space on your phone : 10,000,000+
SD card backup : 1,000,000+
Nokia mobile support : 5,000,000+
File Manager -- Take Command of Your Files Easily : 10,000,000+
Samsung Calculator : 100,000,000+
Clear : 10,000,00

Roland DG Mobile Panel : 1,000+
DG-App : 500+
Digoo·Cloud : 10,000+
DiskDigger photo recovery : 50,000,000+
Automatic Call Recorder : 100,000,000+
Virtual DJ Mixer : 100,000+
Mix Virtual DJ 2018 : 100,000+
Advanced Download Manager : 10,000,000+
GetThemAll Any File Downloader : 5,000,000+
Downloader & Private Browser : 50,000,000+
Download All Files : 10,000,000+
Download Accelerator Plus : 1,000,000+
Download Manager : 10,000,000+
Fast Download Manager : 1,000,000+
Turbo Download Manager (and Browser) : 5,000,000+
Download Blazer : 5,000,000+
Download Manager For Android Free : 100,000+
Turbo Downloader : 5,000,000+
Advanced Download Manager Holo : 1,000,000+
G-Download Manager : 50,000+
TorrDroid - Torrent Downloader : 1,000,000+
Download Manager - File & Video : 1,000,000+
DM airdisk : 1,000+
DM HiDisk : 5,000+
DM airdisk Pro : 1,000+
DM security - Dragon Mobile : 1,000+
DM 24/7 : 1,000+
DM AirDisk HDD : 100+
DM AirDisk NAS : 50+
File Viewer for Android : 1,000,000+
EasyNote Notepad

In [56]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

In [57]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

In [58]:
for app in android_free:
    if app[1] == 'TOOLS' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Moto Display : 10,000,000+
Motorola Alert : 50,000,000+
Motorola Assist : 50,000,000+
Moto Suggestions ™ : 1,000,000+
Moto Voice : 10,000,000+
myMetro : 10,000,000+
File Manager : 50,000,000+
My Telcel : 50,000,000+
Calculator - free calculator, multi calculator app : 10,000,000+
ASUS Sound Recorder : 10,000,000+
iWnn IME for Nexus : 5,000,000+
Samsung Max - Data Savings & Privacy Protection : 10,000,000+
Android TV Remote Service : 1,000,000+
ZenUI Help : 10,000,000+
ZenUI Keyboard – Emoji, Theme : 10,000,000+
Files Go by Google: Free up space on your phone : 10,000,000+
SD card backup : 1,000,000+
Nokia mobile support : 5,000,000+
File Manager -- Take Command of Your Files Easily : 10,000,000+
Clear : 10,000,000+
Phone : 10,000,000+
HTC Lock Screen : 10,000,000+
AT&T Smart Wi-Fi : 10,000,000+
Google app for Android TV : 10,000,000+
Sound Recorder: Recorder & Voice Changer Free : 10,000,000+
Remote Link (PC Remote) : 10,000,000+
HTC Sense Input : 10,000,000+
App vault : 10,000,000+
To

We see similar patterns from the Appstore.
In the top 5 install categories, there are some huge apps with 1 billion installs and a smaller group with 50-100 million installs. It is difficult to compete with such applications, most often there are huge and complex products behind them.

In my opinion good categories for free apps now are:
- Games (Family in Google Store). Although this is an overheated category, it still has a demand. People spend a lot of time there, and simple games are often played by kids who spend a lot of time in the app one way or another.
- Tools. There is a demand for them, there are many small applications with 1 million installs, and it is easier to compete there, although they do not spend much time in such applications.
- Books. There is a demand for a book app as well as for book readers, people spend a lot of time in such apps (reading a book is not a 10 minute task) and creating such an app is not that difficult a task.


We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

## Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.